src/lib.rs

```diff
@@ -4,11 +4,18 @@ use ispc::WeightCollection;
 
 mod ispc;
 
-pub trait ImagePixelFormat {
-    fn num_channel_in_memory(&self) -> usize;
-    fn channel_size_in_bytes(&self) -> usize;
-
-    fn pixel_size(&self) -> usize {
+pub trait ImagePixelFormat: Copy {
+    /// Returns the number of channels that an image of this format would have in memory.
+    /// For example, while a normal map of format [`NormalMapFormat::R8g8TangentSpaceReconstructedZ`] would still have 3 channels when sampled,
```

Comment:
Is it just me that thinks `normalmap` should be one word?

Reply:
Seems like online concensus, as well as in breda, is that `normal map` is two words.

Unless you mean that in this case `normal map` can be misinterpreted, but I think this is too nitty to address.

Reply:
> Unless you mean that in this case `normal map` can be misinterpreted, but I think this is too nitty to address.

Yeah, my brain made a mis-step here while parsing this sentence.